### PR TITLE
add rsm to lockstack

### PR DESCRIPTION
**Pull request by Greg-Griffith**

### CODE REVIEW

File: src/sync.cpp [hunk 1]
```diff
@@ -387,4 +401,102 @@ bool CSharedCriticalSection::try_lock()
 }
 
 
+
+// BU normally CRecursiveSharedCriticalSection is a typedef, but when lockorder debugging is on we need to delete the critical
+// section from the lockorder map
+#ifdef DEBUG_LOCKORDER
+CRecursiveSharedCriticalSection::CRecursiveSharedCriticalSection() : name(NULL), exclusiveOwner(0) {}
+CRecursiveSharedCriticalSection::CRecursiveSharedCriticalSection(const char *n) : name(n), exclusiveOwner(0)
```
Comment (ptschip) [inline src/sync.cpp]: -NULL
+nullptr

**Comment by sickpig** [comment]
@Greg-Griffith since #1720 got merged I'd say we should remove `[DEPENDS 1720]` from PR title

**Comment by Greg-Griffith** [comment]
> @Greg-Griffith since #1720 got merged I'd say we should remove `[DEPENDS 1720]` from PR title

you got it mate

**Comment by ptschip** [comment]
@Greg-Griffith getting a segfault when I run this in debug...just creating a simple csection and locking will cause it.

CRecursiveSharedCriticalSection cs;
RECURSIVEREADLOCK(cs);

**Review by ptschip — APPROVED** [review]
I actually tested that the lockstack updates correctly when locks are added and removed.  Works fine...

**Comment by sickpig** [comment]
> I actually tested that the lockstack updates correctly when locks are added and removed. Works fine...

Wonder if we could find a way to do it in automated way, e.g. unit test or some such, even thou it would be very tricky to do.

**Pull request merged**